File: app/src/main/java/monster/minions/binocularss/Article.java
package monster.minions.binocularss;

public class Article {
    private String title;
    private String date;
    private String url;
    private String source;
    private String author;
    private String text;
    private String description;

    public Article(String title, String date, String url, String source, String author, String text, String description) {
        this.title = title;
        this.date = date;
        this.url = url;
        this.source = source;
        this.author = author;
        this.text = text;
        this.description = description;
    }

    public Article(String title, String url, String description) {
        this.title = title;
        this.url = url;
        this.description = description;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: app/src/main/java/monster/minions/binocularss/Article.kt
package monster.minions.binocularss

import android.os.Parcel
import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 * A dataclass object representing an Article that is part of an RSS Feed.
 *
 * Info on possible tags from https://validator.w3.org/feed/docs/rss2.html
 *
 * @param title The title of the item.
 * @param author Email address of the author of the item.
 * @param link The URL of the item.
 * @param pubDate Indicates when the item was published.
 * @param description The item synopsis.
 * @param content The content of an item, also known as enclosure.
 * @param image An href to a header image.
 * @param audio An href to companion audio.
 * @param video An href to companion video.
 * @param sourceName The source name (name of the feed it came from).
 * @param sourceUrl The source URL (url of the feed it came from).
 * @param categories A list of the appropriate categories (defined by RSS).
 * @param guid A string that uniquely identifies the item.
 * @param source The RSS channel that the item came from.
 */
@Parcelize
data class Article(
    var title: String? = "",
    var author: String? = "",
    var link: String? = "",
    var pubDate: String? = "",
    var description: String? = "",
    var content: String? = "",
    var image: String? = "",
    var audio: String? = "",
    var video: String? = "",
    var sourceName: String? = "",
    var sourceUrl: String? = "",
    var categories: MutableList<String>? = mutableListOf<String>(),
    var guid: String? = "",
    var source: String? = "",
) : Parcelable